libs/features/personalization/preview.js

```diff
@@ -296,10 +343,37 @@ function addHighlightData(manifests) {
       .forEach((el) => (el.dataset.manifestId = manifestName));
   });
 }
-
+export async function saveToMmm() {
+  const data = parseMepConfig();
+  if (data.page.url.includes('/drafts/')) return false;
+  data.activities = data.activities.filter((activity) => {
+    const { url, source } = activity;
+    activity.source = source.filter((item) => item !== 'mep param');
+    return (!!(activity.source?.length && !url.includes('/drafts/')));
+  });
+  data.activities = data.activities.map((activity) => {
+    activity.variantNames = activity.variantNames?.join('||') || '';
+    activity.source = activity.source?.join(',') || '';
+    delete activity.selectedVariantName;
+    return activity;
+  });
+  if (data.page.prefix === US_GEO) data.page.prefix = '';
+  data.page.target = getMetadata('target') || 'off';
+  delete data.page.highlight;
+  return fetch(API_URLS.save, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(data),
+  })
+    .then((response) => {
+      /* c8 ignore next 1 */
+      if (!response.ok) throw new Error('Network response was not ok');
```

Comment:
let's do it like this 
`.then((response) => {
      /* c8 ignore next 1 */
      if (response.ok) {
        return response.json();
      } else {
        const errorMessage = response.text();
         throw new Error(errorMessage || 'Network response was not ok');
      }
    });
    `

Reply:
that way we will pass the actual server text of the error

Reply:
@denlight updated to below to comply with linting rules:
    .then(async (response) => {
      const res = await response.json();
      if (response.ok) return res;
      /* c8 ignore next 1 */
      throw new Error(res.message || 'Network response failed');
    });